src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/Filters/DuplicatedDependenciesSnapshotFilter.cs

```diff
@@ -33,22 +33,43 @@ internal class DuplicatedDependenciesSnapshotFilter : DependenciesSnapshotFilter
             out bool filterAnyChanges)
         {
             filterAnyChanges = false;
-            IDependency resultDependency = dependency;
+            var resultDependency = dependency;
 
-            var matchingDependency = topLevelBuilder.FirstOrDefault(
-                x => !x.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase)
-                     && x.ProviderType.Equals(dependency.ProviderType, StringComparison.OrdinalIgnoreCase) 
-                     && x.Caption.Equals(dependency.Caption, StringComparison.OrdinalIgnoreCase));
-            
-            // if found node with same caption, or if there were nodes with same caption but with Alias already applied
-            var shouldApplyAlias = (matchingDependency == null)
-                ? topLevelBuilder.Any(
-                    x => !x.Id.Equals(dependency.Id)
+            IDependency matchingDependency = null;
+            foreach (var x in topLevelBuilder)
+            {
+                if (!x.Id.Equals(dependency.Id, StringComparison.OrdinalIgnoreCase)
+                     && x.ProviderType.Equals(dependency.ProviderType, StringComparison.OrdinalIgnoreCase)
+                     && x.Caption.Equals(dependency.Caption, StringComparison.OrdinalIgnoreCase))
+                {
+                    matchingDependency = x;
+                    break;
+                }
+            }
+
+            // If found node with same caption, or if there were nodes with same caption but with Alias already applied
+            // NOTE: Performance sensitive, so avoid formatting the Caption with parens if it's possible to avoid it.
```

Comment:
>parens [](start = 81, length = 6)

typo

Reply:
What is the typo you're seeing?  I'm still not seeing it?

Reply:
Ah - you want me to spell out "parentheses" instead of "parens"?